cloudinit/config/cc_ansible.py

```diff
@@ -111,20 +119,18 @@ def install(self, pkg_name: str):
         if not self.is_installed():
             # bootstrap pip if required
             if not which("pip3"):
-                self.distro.install_packages(PIP_PKG)
-            subp(["python3", "-m", "pip", "install", "--user", pkg_name])
+                self.distro.install_packages(self.distro.pip_package_name)
+            self.subp(["python3", "-m", "pip", "install", "--user", pkg_name])
```

Comment:
pip3 and python3 might not be available on every platform.
can't we use the python we are executed under as starting off point?

Reply:
Thanks @igalic for the feedback. How about I replace `python3` with `sys.executable`?

Reply:
👍

Reply:
@igalic: Regarding pip3, I'm not certain what the best approach would be. Maybe something like this?

```python
-            if not which("pip3"):
+            _, stderr = self.subp(
+                [sys.executable, "-m", "pip"], rcs=[0, 1]
+            )
+            if stderr.strip():
```

Reply:
we could do a scoped import, and catch the error, then, based on that install